nixos/modules/tasks/filesystems/zfs.nix

```diff
@@ -58,6 +58,24 @@ let
 
   snapshotNames = [ "frequent" "hourly" "daily" "weekly" "monthly" ];
 
+  importLib = {pool, zpoolCmd, awkCmd, cfgZfs}: ''
+    ready() {
+      state="$("${zpoolCmd}" import | "${awkCmd}" '/pool: +${pool}/ { found = 1 }; /state:/ { if (found == 1) { print $2; exit } }; END { if (found == 0) { print "MISSING" } }')"
+      if [[ "$state" = "ONLINE" ]]; then
+        return 0
+      else
+        echo "Pool ${pool} in state $state, waiting"
+        return 1
+      fi
+    }
+    imported() {
+      "${zpoolCmd}" list "${pool}" >/dev/null 2>/dev/null
+    }
+    import() {
+      "${zpoolCmd}" import -d "${cfgZfs.devNodes}" -N $ZFS_FORCE "${pool}"
```

Comment:
I would recommend to rename `ready` to `zpool-ready`, which takes the pool name as the first argument.
It would be actually more readable if the other functions would be inline, since the commands are not much more complex  then there replacement.

Reply:
Fixed the names.

I think the functions are quite a bit more readable than their contents, but that's not the only reason they exist. It's also to avoid duplicating the code. We might need to change some of this in the future, and it's good not having to change two places.

Note that the import systemd unit and the import-root-pool scripts had already diverged, so that isn't an academic concern.